src/utils/rustc_version.rs

```diff
@@ -19,6 +21,38 @@ pub fn parse_rustc_version<S: AsRef<str>>(version: S) -> Result<String> {
     ))
 }
 
+fn parse_rustc_date<S: AsRef<str>>(version: S) -> Result<Date<Utc>> {
+    static RE: Lazy<Regex> = Lazy::new(|| Regex::new(r" (\d+)-(\d+)-(\d+)\)$").unwrap());
+
+    let cap = RE
+        .captures(version.as_ref())
+        .with_context(|| anyhow!("Failed to parse rustc date"))?;
+
+    let year = cap.get(1).unwrap().as_str();
+    let month = cap.get(2).unwrap().as_str();
+    let day = cap.get(3).unwrap().as_str();
+
+    Ok(Utc.ymd(
+        year.parse::<i32>().unwrap(),
+        month.parse::<u32>().unwrap(),
+        day.parse::<u32>().unwrap(),
+    ))
+}
+
+/// Picks the correct "rustdoc.css" static file depending on which rustdoc version was used to
+/// generate this version of this crate.
+pub fn get_correct_docsrs_style_file(version: &str) -> String {
+    if let Ok(date) = parse_rustc_date(version) {
```

Comment:
Can this return an error instead, so we know we need to fix it instead of it silently failing?
```suggestion
    let data = parse_rustc_date(version)?;
```

Reply:
I did and use `.unwrap()` in two places.